src/adagrams.js

```diff
@@ -1,15 +1,183 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    const letterPool = {
+        A: 9,
+        B: 2,
+        C: 2,
+        D: 4,
+        E: 12,
+        F: 2,
+        G: 3,
+        H: 2,
+        I: 9,
+        J: 1,
+        K: 1,
+        L: 4,
+        M: 2,
+        N: 6,
+        O: 8,
+        P: 2,
+        Q: 1,
+        R: 6,
+        S: 4,
+        T: 6,
+        U: 4,
+        V: 2,
+        W: 2,
+        X: 1,
+        Y: 2,
+        Z: 1,
+    };
+    const letters = [
+        'A',
+        'B',
+        'C',
+        'D',
+        'E',
+        'F',
+        'G',
+        'H',
+        'I',
+        'J',
+        'K',
+        'L',
+        'M',
+        'N',
+        'O',
+        'P',
+        'Q',
+        'R',
+        'S',
+        'T',
+        'U',
+        'V',
+        'W',
+        'X',
+        'Y',
+        'Z',
+    ];
+    const randomLetters = (choices) =>
+        choices[Math.floor(Math.random() * choices.length)];
+    let hand = [];
+    while (hand.length != 10) {
+        let letter = randomLetters(letters);
+        for (const [key, value] of Object.entries(letterPool)) {
+            if (letter == key && value != 0) {
+                letterPool[key] -= 1;
+                hand.push(letter);
+            }
+        }
+    }
+    return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+    input = input.toUpperCase();
+    const guessedLetters = Array.from(input);
+    let booleanList = [];
+    const letterCountGuessedLetters = guessedLetters.reduce((obj, letter) => {
+        //counts the number of letters in the input
+        const count = obj[letter] || 0; //if the letter is not in the object, it will be set to 0
+        return {...obj, [letter]: count + 1 }; //adds the letter to the object
+    }, {});
+    const letterCountLettersInHand = lettersInHand.reduce((obj, letter) => {
+        //counts the number of letters in the hand
+        const count = obj[letter] || 0; //if the letter is not in the object, it will be set to 0
+        return {...obj, [letter]: count + 1 }; //adds the letter to the object
+    }, {});
+    for (const [letter, count] of Object.entries(letterCountGuessedLetters)) {
+        //checks if the letter is in the hand
+        if (count <= letterCountLettersInHand[letter]) {
+            //if the letter is in the hand, true or false will be added to the boolean list
+            booleanList.push(true);
+        } else {
+            booleanList.push(false);
```

Comment:
Rather than pushing false onto a boolean list, we could exit early by returning as soon as we find an invalid count.